fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java

```diff
@@ -152,12 +152,84 @@ protected OptExpression viewBasedRewrite(RewriteContext rewriteContext, OptExpre
                 }
                 mvOptExpr = OptExpression.create(new LogicalFilterOperator(rewrittenPred), mvOptExpr);
             }
-            return rewriteProjection(rewriteContext, queryExprToMvExprRewriter, mvOptExpr);
+            return rewriteProjection(rewriteContext, queryAggOp, queryExprToMvExprRewriter, mvOptExpr);
```

Comment:
for bitmap/hll/percentile, it seems that we should support rollup?

Reply:
yep. `Rollup` is supported in `rewriteAggregationOperatorByRules ` method like `avg` before. 

Here we should support `projection` also.

Reply:
is Rollup with distinct supported in this pr?

Reply:
yep. only count (distinct) with rollup is supported: MV is created by bitmap_union(xxx) like single mv.